pkg/scheduler/framework/cycle_state.go

```diff
@@ -78,36 +77,32 @@ func (c *CycleState) Clone() *CycleState {
 		return nil
 	}
 	copy := NewCycleState()
-	for k, v := range c.storage {
-		copy.Write(k, v.Clone())
-	}
+	c.storage.Range(func(k, v interface{}) bool {
+		copy.storage.Store(k, v.(StateData).Clone())
+		return true
+	})
```

Comment:
No relation to this pr.
Why we ignore the `recordPluginMetrics` in `Clone`.

Reply:
Hmm. Exactly. I also think we need to copy `recordPluginMetrics` as well. 
Otherwise, for example the below part is always false, I guess.
https://github.com/kubernetes/kubernetes/blob/1007f4974d299991c5cb840dd3ce8f1cc58fbdce/pkg/scheduler/framework/runtime/framework.go#L632

I'll create another PR for this.

Reply:
https://github.com/kubernetes/kubernetes/pull/108727